extensions/ql-vscode/src/variant-analysis/gh-api/gh-api-client.ts

```diff
@@ -20,18 +23,38 @@ export async function getCodeSearchRepositories(
   token: CancellationToken,
 ): Promise<string[]> {
   let nwos: string[] = [];
-  const octokit = await credentials.getOctokit();
+  const MyOctokit = Octokit.plugin(throttling);
+  const auth = await credentials.getAccessToken();
+
+  const octokit = new MyOctokit({
+    auth,
```

Comment:
To copy more closely what we do at https://github.com/github/vscode-codeql/blob/134c4405f9c7a2ca5af3c9eaa0a01a8fb86cdc50/extensions/ql-vscode/src/common/vscode/authentication.ts#L35-L38, this should ideally include the `retry` plugin too. Or is that intentionally missed out? The idea of this plugin is that'll automatically retry requests in cases of a 500 error.

```suggestion
    auth,
    retry,
```

Reply:
That was an oversight, will add it now!

Reply:
I don't think we need the retry plugin here, as the throttle plugin is taking over retrying?

Reply:
I believe they deal with different concerns. The `retry` plugin will retry in response to 5xx errors. The `throttle` plugin will retry in response to 429 (Too many requests) errors and understands GitHub's specific messages.

But seems that in the latest version of `provideOctokitWithThrottling` the `retry` plugin is included, so that all looks good to me.